azurerm/internal/services/compute/ssh_keys.go

```diff
@@ -111,3 +112,44 @@ func parseUsernameFromAuthorizedKeysPath(input string) *string {
 
 	return nil
 }
+
+// TODO - leverage "golang.org/x/crypto/ssh" for key validation?
+// ValidateSSHKey performs some basic validation on supplied SSH Keys - Signature and basic key length are evaluated
+func ValidateSSHKey(i interface{}, k string) (warnings []string, errors []error) {
+	permittedKeyTypes := []string{
+		"ssh-rsa",
+	}
+
+	v, ok := i.(string)
+	if !ok {
+		return nil, []error{fmt.Errorf("expected type of %q to be string", k)}
+	}
+
+	if strings.TrimSpace(v) == "" {
+		return nil, []error{fmt.Errorf("expected %q to not be an empty string or whitespace", k)}
+	}
+
+	validSig := ""
+	for _, t := range permittedKeyTypes {
+		if strings.HasPrefix(v, t) {
+			validSig = t
+		}
+	}
+
+	switch validSig {
+	case "ssh-rsa":
+		keyParts := strings.Fields(v)
+		if len(keyParts) > 1 {
+			strLen := len(keyParts[1])
+			if strLen < 371 {
```

Comment:
I don't know enough about SSH keys to see the specifics but I feel like we'd be better to parse the SSH key using the crypto library rather than checking the string length?

Reply:
agreed

Reply:
It's not core go SDK (yet) means using importing `golang.org/x/crypto/ssh`, will rework for that as discussed offline.
NB: I don't believe I've ever seen a case where a 2048bit key in SSH2 notation is anything other than 371 characters.